// @alpacahq/typescript-sdk/esm/api/marketData.js
import { baseURLs } from "../factory/createClient.js";
export const getStocksCorporateActions = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/corporate-actions",
    method: "GET",
    params,
});
export const getLogo = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/logos/:symbol",
    method: "GET",
    params,
});
export const getNews = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/news",
    method: "GET",
    params,
});
export const getStocksMostActives = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/screener/stocks/most-actives",
    method: "GET",
    params,
});
export const getStocksMarketMovers = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/screener/stocks/market-movers",
    method: "GET",
    params,
});
export const getStocksQuotes = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/quotes",
    method: "GET",
    params,
});
export const getStocksQuotesLatest = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/quotes/latest",
    method: "GET",
    params,
});
export const getStocksBars = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/bars",
    method: "GET",
    params,
});
export const getStocksBarsLatest = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/bars/latest",
    method: "GET",
    params,
});
export const getForexRates = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/forex/rates",
    method: "GET",
    params,
});
export const getLatestForexRates = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/forex/latest/rates",
    method: "GET",
    params,
});
export const getStocksSnapshots = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/stocks/snapshots",
    method: "GET",
    params,
});
export const getStocksAuctions = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/auctions",
    method: "GET",
    params,
});
export const getStocksConditions = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: `/v2/stocks/meta/conditions/${params.tickType}`,
    method: "GET",
    params: { tape: params.tape },
});
export const getStocksExchangeCodes = (context) => () => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/meta/exchanges",
    method: "GET",
});
export const getStocksTrades = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/trades",
    method: "GET",
    params,
});
export const getStocksTradesLatest = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v2/stocks/trades/latest",
    method: "GET",
    params,
});
export const getOptionsBars = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/options/bars",
    method: "GET",
    params,
});
export const getOptionsExchanges = (context) => () => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/options/meta/exchanges",
    method: "GET",
});
export const getOptionsSnapshots = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/options/snapshots",
    method: "GET",
    params,
});
export const getOptionsTrades = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/options/trades",
    method: "GET",
    params,
});
export const getOptionsTradesLatest = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/options/trades/latest",
    method: "GET",
    params,
});
export const getCryptoBars = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/crypto/bars",
    method: "GET",
    params,
});
export const getLatestCryptoBars = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: `/v1beta3/crypto/${params.loc}/latest/bars`,
    method: "GET",
    params: {
        symbols: params.symbols,
    },
});
export const getCryptoQuotes = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/crypto/quotes",
    method: "GET",
    params,
});
export const getCryptoQuotesLatest = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: `/v1beta3/crypto/${params.loc}/latest/quotes`,
    method: "GET",
    params: {
        symbols: params.symbols,
    },
});
export const getCryptoSnapshots = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta1/crypto/snapshots",
    method: "GET",
    params,
});
export const getCryptoTrades = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: "/v1beta3/crypto/:loc/trades",
    method: "GET",
    params,
});
export const getCryptoTradesLatest = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: `/v1beta3/crypto/${params.loc}/latest/trades`,
    method: "GET",
    params: {
        symbols: params.symbols,
    },
});
export const getLatestCryptoOrderbooks = (context) => (params) => context.request({
    baseURL: baseURLs.marketData,
    path: `/v1beta3/crypto/${params.loc}/latest/orderbooks`,
    method: "GET",
    params: {
        symbols: params.symbols,
    },
});
