// @alpacahq/typescript-sdk/esm/api/trade.js
export const getAccount = ({ request }) => () => request({
    path: "/v2/account",
});
export const createOrder = ({ request }) => (data) => request({
    path: "/v2/orders",
    method: "POST",
    data,
});
export const getOrder = ({ request }) => ({ order_id }) => request({
    path: `/v2/orders/${order_id}`,
    method: "GET",
});
export const getOrders = ({ request }) => (params) => request({
    path: "/v2/orders",
    method: "GET",
    params,
});
export const replaceOrder = ({ request }) => (data) => request({
    path: `/v2/orders/${data.order_id}`,
    method: "PATCH",
    data,
});
export const cancelOrder = ({ request }) => ({ order_id }) => request({
    path: `/v2/orders/${order_id}`,
    method: "DELETE",
});
export const cancelOrders = ({ request }) => () => request({
    path: "/v2/orders",
    method: "DELETE",
});
export const getPosition = ({ request }) => ({ symbol_or_asset_id }) => request({
    path: `/v2/positions/${symbol_or_asset_id}`,
    method: "GET",
});
export const getPositions = ({ request }) => () => request({
    path: "/v2/positions",
    method: "GET",
});
export const closePosition = ({ request }) => ({ symbol_or_asset_id }) => request({
    path: `/v2/positions/${symbol_or_asset_id}`,
    method: "DELETE",
});
export const closePositions = ({ request }) => () => request({
    path: "/v2/positions",
    method: "DELETE",
});
export const exerciseOption = ({ request }) => ({ symbol_or_contract_id }) => request({
    path: `/v2/positions/${symbol_or_contract_id}/exercise`,
    method: "POST",
});
export const getCalendar = ({ request }) => (params = {}) => request({
    path: "/v2/calendar",
    method: "GET",
    params,
});
export const getClock = ({ request }) => () => request({
    path: "/v2/clock",
});
export const getAsset = ({ request }) => ({ symbol_or_asset_id }) => request({
    path: `/v2/assets/${symbol_or_asset_id}`,
    method: "GET",
});
export const getAssets = ({ request }) => (params = {}) => request({
    path: "/v2/assets",
    method: "GET",
    params,
});
export const getWatchlist = ({ request }) => ({ watchlist_id }) => request({
    path: `/v2/watchlists/${watchlist_id}`,
});
export const getWatchlists = ({ request }) => () => request({
    path: "/v2/watchlists",
});
export const createWatchlist = ({ request }) => (data) => request({
    path: "/v2/watchlists",
    method: "POST",
    data,
});
export const updateWatchlist = ({ request }) => (data) => request({
    path: `/v2/watchlists/${data.watchlist_id}`,
    method: "PATCH",
    data,
});
export const deleteWatchlist = ({ request }) => ({ watchlist_id }) => request({
    path: `/v2/watchlists/${watchlist_id}`,
    method: "DELETE",
});
export const getPortfolioHistory = ({ request }) => (params) => request({
    path: "/v2/account/portfolio/history",
    method: "GET",
    params,
});
export const getConfigurations = ({ request }) => () => request({
    path: "/v2/account/configurations",
});
export const updateConfigurations = ({ request }) => (data) => request({
    path: "/v2/account/configurations",
    method: "PATCH",
    data,
});
export const getActivity = ({ request }) => ({ activity_type }) => request({
    path: `/v2/account/activities/${activity_type}`,
    method: "GET",
});
export const getActivities = ({ request }) => () => request({
    path: "/v2/account/activities",
    method: "GET",
});
export const getOptionsContract = ({ request }) => ({ symbol_or_contract_id }) => request({
    path: `/v2/options/contracts/${symbol_or_contract_id}`,
    method: "GET",
});
export const getOptionsContracts = ({ request }) => (params) => request({
    path: "/v2/options/contracts",
    method: "GET",
    params,
});
export const getCorporateAction = ({ request }) => ({ id }) => request({
    path: `/v2/corporate_actions/announcements/${id}`,
    method: "GET",
});
export const getCorporateActions = ({ request }) => (params) => request({
    path: "/v2/corporate_actions/announcements",
    method: "GET",
    params,
});
export const getCryptoWallet = ({ request }) => ({ asset }) => request({
    path: `/v2/wallets/${asset}`,
    method: "GET",
});
export const getCryptoWallets = ({ request }) => () => request({
    path: "/v2/wallets",
    method: "GET",
});
export const getFeeEstimate = ({ request }) => (params) => request({
    path: "/v2/wallets/fees/estimate",
    method: "GET",
    params,
});
export const getCryptoTransfer = ({ request }) => ({ transfer_id }) => request({
    path: `/v2/wallets/transfers/${transfer_id}`,
    method: "GET",
});
export const getCryptoTransfers = ({ request }) => (params) => request({
    path: "/v2/wallets/transfers",
    method: "GET",
    params,
});
export const createCryptoTransfer = ({ request }) => (data) => request({
    path: "/v2/wallets/transfers",
    method: "POST",
    data,
});
export const getCryptoWhitelistedAddress = ({ request }) => (params) => request({
    path: "/v2/wallets/whitelists",
    method: "GET",
    params,
});
export const getCryptoWhitelistedAddresses = ({ request }) => () => request({
    path: "/v2/wallets/whitelists",
    method: "GET",
});
export const requestCryptoWhitelistedAddress = ({ request }) => (data) => request({
    path: "/v2/wallets/whitelists",
    method: "POST",
    data,
});
export const removeCryptoWhitelistedAddress = ({ request }) => ({ whitelisted_address_id }) => request({
    path: `/v2/wallets/whitelists/${whitelisted_address_id}`,
    method: "DELETE",
});
